components/brave_welcome_ui/components/select-browser/index.tsx

```diff
@@ -78,7 +78,10 @@ function BrowserItemButton (props: BrowserItemButtonProps) {
 function SelectBrowser () {
   const { browserProfiles, currentSelectedBrowser, setCurrentSelectedBrowser, setViewType, incrementCount, scenes } = React.useContext(DataContext)
   const browserTypes = getUniqueBrowserTypes(browserProfiles ?? [])
-
+  WelcomeBrowserProxyImpl.getInstance().getDefaultBrowser().then(
```

Comment:
React functional components are meant to be pure. Any async calls like this will block the render and it will also trigger on subsequent re-render.

Please trigger this call from a React's lifecycle.

```js

React.useEffect(() => {
// call here
}, [])
```